### ISSUE TITLE
Rewrite default plotting system

### ISSUE THREAD

**Issue by ChrisRackauckas** [SciML/SciMLBase.jl#11]
As noted in https://github.com/SciML/SciMLBase.jl/pull/9/files#issuecomment-770289578, the default plotting system is somewhat of a mess because it's just faking functionality that we now have, like:

1. Indexing via symbols
2. Observed variables
3. Independent variable symbol handling

It should get a refactor to greatly decrease the complexity by using these aspects of the system.

**Comment by scheidan** [comment]
I've just observed that currently the labels in an ODE plot are not displayed correctly: the legend it repeats three times `("A", "B", "C")`.

Small example with `SciMLBase v1.7.3`:
```Julia
function f(du, u,p,t)
    k_u, k_m, k_em_p, k_em_m = p
    du[1] = 0.0
    du[2] = k_u*u[1] - k_em_p*u[2] - k_m*u[2]
    du[3] = k_m*u[2] - k_em_m*u[3]
end

u0 = [10, 0, 0]
p = (k_u=1, k_m=0.2, k_em_p=0.9, k_em_m=0.1)
prob = ODEProblem(f, u0, (0, 10.0), p)
sol = solve(prob, Tsit5())

plot(sol, labels=("A", "B", "C"))
```

**Comment by ChrisRackauckas** [comment]
`plot(sol, labels=["A"  "B"  "C"])` is how Plots.jl would do it IIRC?

**Comment by scheidan** [comment]
Yes, I think so. that's also how it is done in this ODE tutorial:
 https://diffeq.sciml.ai/latest/tutorials/ode_example/#Defining-Parameterized-Functions

**Comment by lamorton** [comment]
This would probably also resolve [#1010](https://github.com/SciML/ModelingToolkit.jl/issues/1018).
